source/MaterialXTest/MaterialXGenShader/GenShaderUtil.cpp

```diff
@@ -691,6 +691,13 @@ void ShaderGeneratorTester::validate(const mx::GenOptions& generateOptions, cons
         findLights(doc, _lights);
         registerLights(doc, _lights, context);
 
+
+        // Add any userdata if required
+        if (enableUserData)
+        {
+            addUserData(context);
```

Comment:
Will this keep on adding user data if you run validate more than once?
We can do this for minimum and maximum uniform exposure configurations.

Reply:
context is local in validate(). adduserdata() basically does a push that does not duplicate.